misk-redis/src/main/kotlin/misk/redis/JedisPoolWithMetrics.kt

```diff
@@ -0,0 +1,78 @@
+package misk.redis
+
+import org.apache.commons.pool2.PooledObject
+import redis.clients.jedis.Jedis
+import redis.clients.jedis.JedisFactory
+import redis.clients.jedis.JedisPool
+import redis.clients.jedis.JedisPoolConfig
+import redis.clients.jedis.Protocol
+
+internal class JedisPoolWithMetrics(
+  private val metrics: RedisClientMetrics,
+  poolConfig: JedisPoolConfig,
+  replicationGroupConfig: RedisReplicationGroupConfig,
+  ssl: Boolean = true,
+  requiresPassword: Boolean = true,
+) : JedisPool(
+  poolConfig,
+  JedisFactoryWithMetrics(metrics, replicationGroupConfig, ssl, requiresPassword)
+) {
+  init {
+    metrics.maxTotalConnectionsGauge.set(this.maxTotal.toDouble())
+    metrics.maxIdleConnectionsGauge.set(this.maxIdle.toDouble())
+    setActiveIdleConnectionMetrics()
+  }
+
+  override fun getResource(): Jedis {
+    return super.getResource().also {
+      setActiveIdleConnectionMetrics()
+    }
+  }
+
+  override fun returnBrokenResource(resource: Jedis?) {
+    super.returnBrokenResource(resource).also {
+      setActiveIdleConnectionMetrics()
+    }
+  }
+
+  override fun returnResource(resource: Jedis?) {
+    super.returnResource(resource).also {
+      setActiveIdleConnectionMetrics()
+    }
+  }
+
+  private fun setActiveIdleConnectionMetrics() {
+    metrics.activeConnectionsGauge.set(this.numActive.toDouble())
+    metrics.idleConnectionsGauge.set(this.numIdle.toDouble())
+  }
+
+  private class JedisFactoryWithMetrics(
+    private val metrics: RedisClientMetrics,
+    replicationGroupConfig: RedisReplicationGroupConfig,
+    ssl: Boolean = true,
+    requiresPassword: Boolean = true,
+  ) : JedisFactory(
+    /* host = */ replicationGroupConfig.writer_endpoint.hostname,
```

Comment:
Why are the named parameters commented out? Why not use named parameters, especially with so many constructor parameters?

Reply:
Maybe the Java Jedis library doesn't have them generated hmmm

Reply:
Java doesn't have named parameters :( This is the best we can do!